src/main/java/Library/Main.java

```diff
@@ -1,25 +1,48 @@
 package Library;
 
-import Data.BookHandler;
+import Data.UserManager;
 import Interface.Menu;
 
+import java.util.InputMismatchException;
+
 public class Main {
     public static void main(String[] args) {
+
+        UserManager userManager = new UserManager();
+
         Library library = new Library();
         Menu menu = new Menu();
 
 
-        while (true) {
-            Menu.showMainMenu();
-            int choice = Menu.getUserInput();
 
-            if (choice == 1){
-                System.out.println(library.getAllBooks());
-            } else if (choice == 2){
-                String title = Menu.getBookInfo("Please enter a title");
-                String author = Menu.getBookInfo("Please enter an author");
-                library.addBook(new Book(title, author));
+
+
+            if (userManager.login(Menu.getUserReply("Please enter your username.")
+                    , Menu.getUserReply("Please enter password."))) {
+                try {
+                while (true) {
+                    Menu.showMainMenu();
+                    int choice = Menu.getUserInput();
+
+                    if (choice == 1) {
+                        System.out.println(library.getAllBooks());
+                    } else if (choice == 2) {
+                        String title = Menu.getUserReply("Please enter a title");
+                        String author = Menu.getUserReply("Please enter an author");
+                        library.addBook(new Book(title, author));
+                    } else if (choice == 3) {
+                        break;
+                    } else if (choice == 4) {
+                        break;
+                    } else if (choice == 5) {
+                        break;
+                    } else {
+                        throw new IllegalArgumentException("Invalid choice entered");
```

Comment:
It's ok to use standard exception, but even better it would be to create your own exception ie. WrongMenuChoiceExceprion. Go and try! :)

Reply:
Done but I don't understand why wasn't there another catch before and IllegalArgumentException managed to print the message on it own, while after creating new class and extending it of an Exception, I must do yet another catch.